tests/run-with-compiler/reflect-typeChecks/test_2.scala

```diff
@@ -0,0 +1,15 @@
+object Test {
+  import scalatest._
+
+  trait Eq[T]
+  implicit val eq: Eq[Int] = new Eq[Int] {}
+
+  implicit class AnyOps[T](x: T) {
+    def === (y: T)(implicit c: Eq[T]) = x == y
+  }
+
+  def main(args: Array[String]): Unit = {
+    assertCompiles("5 === 5")
+    assertNotCompile("5.6 === 7.7")
```

Comment:
You should add many more tests that depend on the context in different ways. For example, local val/def/type defined in scope. Such as
```scala
val x: Int = 5
assertCompiles("x + 3")
assertNotCompiles("y + 3")
import scala.util.Left
assertCompiles("Left(3)")
assertNotCompiles("Rigth(3)")
```

Reply:
I would also add some code that does not parse